src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonService.kt

```diff
@@ -101,28 +101,17 @@ class PrisonPersonService(
       throw NotFoundException("No offender found for $offenderNo")
     }
 
-    val bookings = bookingRepository.findAllByOffenderNomsId(offenderNo)
-    if (bookings.isEmpty()) {
-      throw NotFoundException("No bookings found for $offenderNo")
-    }
-
-    val latestBookingWithPhysicalAttributes =
-      bookings
-        .filter { it.physicalAttributes.isNotEmpty() }
-        // a null end date is considered the latest, matching end dates (e.g. both null) should pick the lowest booking sequence
-        .maxWithOrNull(compareBy({ it.bookingEndDate ?: LocalDateTime.MAX }, { -it.bookingSequence }))
-
-    val physicalAttributes = latestBookingWithPhysicalAttributes
-      ?.physicalAttributes
-      ?.maxBy { it.modifyDatetime ?: it.createDatetime }
-
-    return physicalAttributes?.let {
-      PrisonPersonReconciliationResponse(
-        offenderNo = offenderNo,
-        height = physicalAttributes.getHeightInCentimetres(),
-        weight = physicalAttributes.getWeightInKilograms(),
-      )
-    }
+    val latestBooking = bookingRepository.findLatestByOffenderNomsId(offenderNo)
+      ?: throw NotFoundException("No bookings found for $offenderNo")
+
+    return latestBooking.physicalAttributes.minByOrNull { it.id.sequence }
+      ?.let {
+        PrisonPersonReconciliationResponse(
+          offenderNo = offenderNo,
+          height = it.getHeightInCentimetres(),
+          weight = it.getWeightInKilograms(),
+        )
```

Comment:
We were previously attempting to apply the same business rules as DPS when determining which is the "active" height and weight. Which is a bit silly because the reconciliation is meant to show that the same values are shown in DPS and NOMIS.

This does mean that the first reconciliation will highlight a lot of errors - mainly where heights/weight haven't been copied from old bookigns in NOMIS - but DPS are aware of this and plan to do a one-off update in NOMIS to align it with the DPS values.